slang-checks/src/main/java/com/sonarsource/slang/checks/HardcodedCredentialsCheck.java

```diff
@@ -0,0 +1,116 @@
+import com.sonarsource.slang.api.Tree;
+import com.sonarsource.slang.api.VariableDeclarationTree;
+import com.sonarsource.slang.checks.api.CheckContext;
+import com.sonarsource.slang.checks.api.InitContext;
+import com.sonarsource.slang.checks.api.SlangCheck;
+import java.util.List;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import javax.annotation.Nullable;
+import org.sonar.check.Rule;
+import org.sonar.check.RuleProperty;
+
+@Rule(key = "S2068")
+public class HardcodedCredentialsCheck implements SlangCheck {
+
+  private static final String DEFAULT_VALUE = "password,passwd,pwd";
+
+  @RuleProperty(
+    key = "credentialWords",
+    description = "Comma separated list of words identifying potential credentials",
+    defaultValue = DEFAULT_VALUE)
+  public String credentialWords = DEFAULT_VALUE;
+
+  private List<Pattern> variablePatterns;
+  private List<Pattern> literalPatterns;
+
+  @Override
+  public void initialize(InitContext init) {
+    init.register(AssignmentExpressionTree.class, (ctx, tree) -> {
+      Tree leftHandSide = tree.leftHandSide();
+      if (leftHandSide instanceof IdentifierTree && tree.statementOrExpression() instanceof StringLiteralTree) {
+        getPasswordVariableName(((IdentifierTree) leftHandSide).name())
+          .ifPresent(passwordVariableName -> report(ctx, leftHandSide, passwordVariableName));
+      }
+    });
+
+    init.register(VariableDeclarationTree.class, (ctx, tree) -> {
+      if (tree.initializer() instanceof StringLiteralTree) {
+        getPasswordVariableName(tree.identifier().name())
+          .ifPresent(passwordVariableName -> report(ctx, tree.identifier(), passwordVariableName));
+      }
+    });
+
+    init.register(StringLiteralTree.class, (ctx, tree) -> literalPatterns()
+      .map(pattern -> pattern.matcher(tree.content()))
+      .filter(Matcher::find)
+      .map(matcher -> matcher.group(1))
+      .forEach(credential -> report(ctx, tree, credential)));
+
+  }
+
+  private static void report(CheckContext ctx, Tree tree, String matchName) {
+    String message = String.format("'%s' detected in this expression, review this potentially hardcoded credential.", matchName);
+    ctx.reportIssue(tree, message);
+  }
+
+  private Optional<String> getPasswordVariableName(@Nullable String name) {
+    if (name == null) {
+      return Optional.empty();
+    }
+    return variablePatterns()
+      .map(pattern -> pattern.matcher(name))
+      .filter(Matcher::find)
+      .map(matcher -> matcher.group(1))
+      .findAny();
+  }
+
+  private Stream<Pattern> variablePatterns() {
+    if (variablePatterns == null) {
+      variablePatterns = toPatterns("");
+    }
+    return variablePatterns.stream();
+  }
+
+  private Stream<Pattern> literalPatterns() {
+    if (literalPatterns == null) {
+      literalPatterns = toPatterns("=\\S.");
```

Comment:
This is to avoid FP when a string `"password= "` used for logging (https://jira.sonarsource.com/browse/SUPPORT-4933).
But it actually made me realize that the pattern was wrong: it should be `=\\S` without the dot, since the double quote is not part of the string literal content already.
Updated in 427fbbc489030925e550a9e7fbe7ea767dd2ff41